### PR TITLE
🐛 Source Quickbooks: use number data type for decimal fields

### LINKED ISSUE #4292: Source Quickbooks: Number data types are being transformed to strings
## Expected Behavior
According to [docs](https://docs.airbyte.io/integrations/sources/quickbooks) supported streams should leave number fields types as numbers in output schema.

## Current Behavior
Most of number fields are being transformed to strings.
It's probably caused by these fields  in `quickbooks-tap` being defined as  
```
      "format": "singer.decimal",
      "type": [
        "null",
        "string"
]
```

## Steps to Reproduce
1. Create quickbooks source. Quickbooks lets you create free trial account with demo organisation already holding some test data.
2. Create connection between quickbooks source and any destination.
3. Check the source schema, e. g. `Invoices` stream and `TotalAmt` field. It's type is set to string, when it should be a number.

## Severity of the bug for you
Medium

## Airbyte Version
0.26.2-alpha

## Connector Version (if applicable)
0.1.1

### PR DESCRIPTION
**Pull request by vitaliizazmic**
## What
Using number data type for decimal fields instead string

## How
Created fork of singer tap, changed schemes. Also migrated Source to Airbyte CDK.

Closes #4292   

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in the connector's spec
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md` contains a reference to the new connector
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

### CODE REVIEW

**Review by gaart — APPROVED** [review]

**Comment by vitaliizazmic** [comment]
test connector=source-quickbooks-singer

File: airbyte-integrations/connectors/source-quickbooks-singer/build.gradle [hunk 1]
```diff
@@ -1,21 +1,14 @@
...
-    specPath = "source_quickbooks_singer/spec.json"
-    configPath = "secrets/config.json"
-    configuredCatalogPath = "sample_files/configured_catalog.json"
-}
-
-
 dependencies {
-    implementation files(project(':airbyte-integrations:bases:base-standard-source-test-file').airbyteDocker.outputs)
+    implementation files(project(':airbyte-integrations:bases:source-acceptance-test').airbyteDocker.outputs)
     implementation files(project(':airbyte-integrations:bases:base-singer').airbyteDocker.outputs)
```
Comment (keu) [inline airbyte-integrations/connectors/source-quickbooks-singer/build.gradle:13]: don't need docker image for singer anymore, we are using cdk PYPI package, which already includes singer wrappers
```suggestion
```

File: airbyte-integrations/connectors/source-quickbooks-singer/acceptance-test-config.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      status: "failed"
+  discovery:
+    - config_path: "secrets/config.json"
+  basic_read:
+    - config_path: "secrets/config.json"
+      configured_catalog_path: "integration_tests/configured_catalog.json"
+      empty_streams: ["budgets", "classes", "departments", "payment_methods", "transfers", "vendor_credits"]
+#  incremental:
+#  This test is commented out, because singer tap expects LastUpdatedTime for each stream in state.
+#  But records contain field MetaData with nested LastUpdatedTime. It causes test error.
```
Comment (keu) [inline airbyte-integrations/connectors/source-quickbooks-singer/acceptance-test-config.yml:20]: not true, SAT supports absolute paths

File: airbyte-integrations/connectors/source-quickbooks-singer/acceptance-test-docker.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env sh
```
Comment (keu) [inline airbyte-integrations/connectors/source-quickbooks-singer/acceptance-test-docker.sh:1]: could you update it in accordance to latest file in from the templates?

**Review by vovavovavovavova — APPROVED** [review]

**Review by sherifnada — APPROVED** [review]

**Review by sherifnada — APPROVED** [review]

**Pull request merged**